### PR TITLE
Feature: Search improvements

### PR DESCRIPTION
**Pull request by miagilepner**

### CODE REVIEW

File: lib/GetStream/StreamChat/Client.php [hunk 1]
```diff
@@ -817,13 +817,28 @@ public function search($filterConditions, $query, $options=null)
...
         }
 
+        $sortFields = [];
+        if (array_key_exists('sort', $options)) {
+            $sort = $options['sort'];
+            foreach ($sort as $k => $v) {
+                $sortFields[] = ["field" => $k, "direction" => $v];
+            }
+        }
+        $options['sort'] = $sortFields;
```
Comment (ferhatelmas) [inline lib/GetStream/StreamChat/Client.php:841]: not sure if needed, but might be better not to mutate given param

Comment (miagilepner) [reply lib/GetStream/StreamChat/Client.php:841]: tested it and it won't mutate the original. array assignment will copy an array and not modify the original, see the Example #15 Recursive and multi-dimensional arrays here: https://www.php.net/manual/en/language.types.array.php

Comment (ferhatelmas) [reply lib/GetStream/StreamChat/Client.php:841]: Tested too. Quite reverse of intuition but seems it always copies and reference is explicit.

```
<?php

function doSomething($options) {
    $sortFields = [];
    if (array_key_exists('sort', $options)) {
        $sort = $options['sort'];
        foreach ($sort as $k => $v) {
            $sortFields[] = ["field" => $k, "direction" => $v];
        }
    }
    $options['sort'] = $sortFields;
}


$options = ["sort" => [["created_at" => -1]]];


echo("before: ");
print_r($options);
doSomething($options);
echo("after: ");
print_r($options);
```

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Pull request merged**